### PR TITLE
config: added config.has() and allow config.get() to be called with no args

### PR DESCRIPTION
**Pull request by Rugvip**
Had enough of awkward `config.getOptionalConfig(...) !== undefined` workarounds, let's add `config.has(key)` 😅 

Also allows `config.get()` to be called without args to serialize the entire config view.

### CODE REVIEW

File: docs/reference/utility-apis/StorageApi.md [hunk 1]
```diff
@@ -181,6 +181,6 @@ export type Subscription = {
 </pre>
 
 Defined at
-[packages/core-api/src/types.ts:33](https://github.com/spotify/backstage/blob/53a229ea7576b1432835e54e41e0b9526038afa4/packages/core-api/src/types.ts#L33).
+[packages/core-api/src/types.ts:33](https://github.com/spotify/backstage/blob/f8780ff32509d0326bc513791ea60846d7614b34/packages/core-api/src/types.ts#L33).
```
Comment (freben) [inline docs/reference/utility-apis/StorageApi.md:184]: so uh, a couple unrelated changes in here :)

**Review by freben — APPROVED** [review]

File: docs/reference/utility-apis/StorageApi.md (hunk 1 above)
Comment (Rugvip) [reply docs/reference/utility-apis/StorageApi.md:184]: Yeah started trying to stage only changed stuff, but there were some line number changes so figured it's nice to have those in sync with master. At that point I cba and just added it all in a separate commit :p
